doc/specs/#194 - PWA Support.md

```diff
@@ -0,0 +1,122 @@
+---
+author: Amrutha Srinivasan @amrutha95
+created on: 2020-08-19
+last updated: 2020-08-19
+issue id: 164
+---
+
+# Spec Title
+
+For [#164](https://github.com/microsoft/winget-cli/issues/164)
+
+## Abstract
+
+This feature adds PWA (Progressive Web Application) support to winget. The goal is to allow users to install and maintain PWAs just like they would any other type of application on winget. As a first step, this would be available as an "experimental" feature to make it available to the community and receive feedback.
+
+## Inspiration
+
+PWA adoption is extremely important in the Windows app ecosystem. It is also crucial that they are treated as first class citizens in the app ecosystem, just like any native application. Providing PWA support for winget is key to achieving this goal.
+
+## Solution Design
+
+### System requirements
+
+Windows 10 v2004 or newer  
+Edge latest Canary build
+
+### PWA YAML manifest
+
+The "InstallerType" field in the YAML manifest file specifies the type of the application. A new InstallerType "PWA" will be added to show that a given application is a PWA.
+
+Each PWA would have its own manifest YAML file. A sample manifest file would look like this:
+
+```Id: PWATest.TestInstaller
+Version: 1.0.0.0
+Name: AppInstaller FinTimes
+Publisher: Microsoft Corporation
+AppMoniker: ft
+License: Test
+InstallerType: PWA
+MinOSVersion: 10.0.0.0
+Installers:
+- Arch: x64
+  Url: https://app.ft.com
+  InstallerType: PWA
+  ManifestVersion: 0.1.0
+```
+
+### Package generation
+
+A POST call is made to a web service (https://pwabuilder-win-chromium-platform.centralus.cloudapp.azure.com) with a JSON body containing the URL to the PWA. The web service packages the PWA and its assets into an unsigned MSIX. An example of the JSON body would be :
+
+```
+{
+    "url": "https://app.ft.com"
+}
+```
+
+There are many more optional parameters that can be passed to this service. For more information, refer to the test website linked in the Resources section.
+
+### Package installation
+
+The unsigned MSIX package and any other dependency packages are added for the current user and the application is silently installed on the user's system. The installation is done by calling the Windows API AddPackageByUriAsync() using the specific deployment options that allow unsigned packages.
+
+### Flow of the install process
+
+1. User types the install command, eg. `winget install fintimes` (For the experimental version, the manifest will be passed as an argument)

```

Comment:
In order to have this command work, either the Name or AppMoniker needs to be an exact match.